Source/Bogus/IRuleSet.cs

```diff
@@ -23,6 +23,11 @@ public interface IRuleSet<T> where T : class
       /// </summary>
       Faker<T> RuleFor<TProperty>(Expression<Func<T, TProperty>> property, Func<Faker, TProperty> setter);
 
+      /// <summary>
+      /// Creates a rule for a property and providing access to the instance being generated.
+      /// </summary>
+      Faker<T> RuleFor<TProperty>(Expression<Func<T, TProperty>> property, Func<T, TProperty> setter);
+
```

Comment:
My current observation of this change is that adding this method to `IRuleSet<T>` causes a slue of source-code breaking changes, not just for us, but for others too. Most of the breaking changes seem to be "ambiguous" call compiler errors that the compiler is having a hard time resolving.

If your ultimate goal with these changes is to make the compiler automatically resolve the overload:
* `.RuleFor(o => o.OrderId, f => orders++);` // captured variable
* `.RuleFor(o => o.OrderId, f => f.Random.Int());` // faker facade
* `.RuleFor(o => o.OrderId, o => o.CreatedDate.Ticks);` // order object

You could probably use an extension method in your own project like the following example:

```csharp
public static class ExtensionsForFakerOfT
{
   /// <summary>
   /// Creates a rule for a property and providing access to the instance being generated.
   /// </summary>
   public static Faker<T> RuleFor<T, TProperty>(this Faker<T> faker, Expression<Func<T, TProperty>> property, Func<T, TProperty> setter)
      where T : class
   {
      return faker.RuleFor(property, (f, t) => setter(t));
   }
}
```
This workaround would be readily available to you and we wouldn't break existing source code in the wild.